### PR TITLE
Fixing bug with "get_dataframe" when certain column name formats are used in the original input data

### PR DESCRIPTION
**Pull request by kyle-wilt-waters**

### CODE REVIEW

**Comment by awcurtin** [comment]
For clarification, this is what we were seeing with `ScaleTorqBlockWave`, right? As far as I can tell, that one gets mapped to snake case because it happens to be Pascal-case in `pascal_to_snake`. Everything else fails the `not name.isidentifier()` check, so they just pass through.

**Comment by kyle-wilt-waters** [comment]
@awcurtin More or less yes. The main issue is that the code was using converted names for one part and unconverted names (kind of) for another. It only became a problem when it actually encountered a situation where it had to use the snake case conversion.

**Comment by stuartncook** [comment]
@kyle-wilt-waters, can you cut down the example data file so it's still valid but has the minimum needed to reproduce the previous error?

File: pyproject.toml [hunk 1]
```diff
@@ -58,5 +58,4 @@ exclude = '''
 [[tool.poetry.source]]
 name = "pypi-test"
 url = "https://test.pypi.org/simple/"
-default = false
-secondary = true
\ No newline at end of file
+priority = "explicit"
```
Comment (stuartncook) [inline pyproject.toml:61]: Is this to work with a later version of Poetry? If so, can we recommend using a version of poetry in CONTRIBUTING.md?

**Comment by kyle-wilt-waters** [comment]
@stuartncook What is your specific concern? Just the size?

File: pyproject.toml (hunk 1 above)
Comment (kyle-wilt-waters) [reply pyproject.toml:61]: I don't know what exactly was barking at me. It came from the commit hooks.

**Comment by kyle-wilt-waters** [comment]
Also bear in mind, the test isn't just to test that bug, it's to test the function in a general way which covers the bug since there are no tests for it before.

**Comment by stuartncook** [comment]
> @stuartncook What is your specific concern? Just the size?

Yes, just the size. How about just dropping the number of rows to 2 or something?

File: pyproject.toml (hunk 1 above)
Comment (stuartncook) [reply pyproject.toml:61]: Actually, let's just drop pypi-test from the sources, it's not needed

Comment (stuartncook) [reply pyproject.toml]: Removed it myself...

**Review by stuartncook — APPROVED** [review]

**Pull request merged**